### PR TITLE
Retry smoke tests

### PR DESCRIPTION
**Pull request by JordonPhillips**
The smoke tests sometimes hit throttling hard enough that they fail,
particualrly when we run them in succession. This adds some retry with
delays on top.

### CODE REVIEW

File: tests/integration/test_smoke.py [hunk 1]
```diff
@@ -115,12 +116,18 @@
...
 
-def _aws(command_string):
+def _aws(command_string, max_attempts=1, delay=5, expected_rc=0):
     service = command_string.split()[0]
     env = None
     if service in REGION_OVERRIDES:
         env = os.environ.copy()
         env['AWS_DEFAULT_REGION'] = REGION_OVERRIDES[service]
+
+    for i in range(max_attempts - 1):
```
Comment (42B) [inline tests/integration/test_smoke.py]:
```python
for _ in range(max_attempts - 1):  # ?
```

File: tests/integration/test_smoke.py [hunk 2]
```diff
@@ -115,12 +116,18 @@
 }
 
 
-def _aws(command_string):
+def _aws(command_string, max_attempts=1, delay=5, expected_rc=0):
```
Comment (kyleknap) [inline tests/integration/test_smoke.py]: I wonder if there is a better name we can use for `expected_rc`? Usually, I expect that it is making an assertion if an `expected_rc` is provided, but the value is being used to determine if we need a retry. So maybe something like `target_rc` or `desired_rc` that indicates we will stop making attempts until we hit that rc?

**Comment by kyleknap** [comment]
Looks good. Just had a small comment.

File: tests/integration/test_smoke.py (hunk 2 above)
Comment (JordonPhillips) [reply tests/integration/test_smoke.py]: sure

**Pull request merged**